src/main/java/org/sopt/lequuServer/domain/book/service/BookService.java

```diff
@@ -27,4 +33,51 @@ public BookCreateResponseDto createBook(Book book, Member member) {
 
         return BookCreateResponseDto.of(createdBook);
     }
+
+    // 정해진 기간의 레큐노트를 가져와 많이 부착된 순서로 레큐북 정렬하는 로직
+    public List<Book> sortBooksDesc(LocalDateTime startDate, LocalDateTime endDate) {
+        return bookRepository.findBooksMonth(startDate, endDate)
+            .stream()
+            .sorted(Comparator.comparingInt(book -> -book.getNotes().size())) // -를 붙여 내림차순 정렬
+            .collect(Collectors.toList());
+    }
+
+    // 최근 3달을 확인하는데 최근 순으로 먼저 인기레큐북에 추가할 수 있는 로직
+    // 현재를 0이라 하면 -1 ~ 0(최근 한달), -2 ~ -1(지지난달 부터 지난달), -3 ~ -2(지지지난달 부터 지지난달)
+    public List<Book> getPopularBooks() {
+        List<Book> popularBooks = new ArrayList<>();
+        LocalDateTime today = LocalDateTime.now();
+
+        // 최근 3달 확인
+        for (int i = 0; i < 3; i++) {
+            LocalDateTime endDate = today.minusMonths(i);
+            LocalDateTime startDate = endDate.minusMonths(1);
+            List<Book> books = sortBooksDesc(startDate, endDate);
+
+            if (popularBooks.size() + books.size() <= 6) {  // 만약 현재있는 인기북과 가져온 북이 6이하면 인기북에 가져온 것 다 추가
+                popularBooks.addAll(duplicateBook(books, popularBooks));
+            } else {
+                int remain = 6 - popularBooks.size();
+                popularBooks.addAll(duplicateBook((books.subList(0, remain)), popularBooks));
+            } // 예를 들어 인기북에 2개가 있으면 4개만 더 추가하면 되니까 가져온 북의 인덱스 0부터 3까지만 가져오도록 함
+
+            if (popularBooks.size() >= 6 || i == 2) {
```

Comment:
```suggestion
                popularBooks.addAll(duplicateBook(books, popularBooks));
                continue;
            } 
            
            int remain = 6 - popularBooks.size();
            popularBooks.addAll(duplicateBook((books.subList(0, remain)), popularBooks));
            break;
            // 예를 들어 인기북에 2개가 있으면 4개만 더 추가하면 되니까 가져온 북의 인덱스 0부터 3까지만 가져오도록 함
```
[P2] 클린 코드를 지향하기 위해 else 예약어는 가급적 안쓰는 편이 좋습니다!
이에 더하여 가져온 레큐 북이 6개가 넘어가는 case는 이 반복문 내에서 한번만 발생하는 상황이니 break문을 통해 분기처리를 하면 조건문이나 코드 스타일 부분에서 최적화가 가능할 것 같습니다 ~~

Reply:
오호 좋은 조언 감사함당 !